src/main/java/nl/inl/blacklab/search/Hit.java

```diff
@@ -29,7 +29,7 @@
  * This class has public members for the sake of efficiency; this makes a non-trivial difference
  * when iterating over hundreds of thousands of hits.
  */
-public class Hit implements Comparable<Hit>, Cloneable {
+public final class Hit implements Comparable<Hit>, Cloneable {
```

Comment:
Making the class final only means you can't extend from it. Is that what you want?